src/main/java/org/broadinstitute/hellbender/utils/spark/RangePartitionCoalescer.java

```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.hellbender.utils.spark;
+
+import org.apache.spark.Partition;
+import org.apache.spark.rdd.PartitionGroup;
+import org.apache.spark.rdd.RDD;
+import scala.collection.JavaConversions;
+import scala.collection.Seq;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+
+class RangePartitionCoalescer implements PartitionCoalescer, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private List<Integer> maxEndPartitionIndexes;
+
+    public RangePartitionCoalescer(List<Integer> maxEndPartitionIndexes) {
+        this.maxEndPartitionIndexes = maxEndPartitionIndexes;
+    }
+
+    @Override
+    public PartitionGroup[] coalesce(int maxPartitions, RDD<?> parent) {
+        if (maxPartitions != parent.getNumPartitions()) {
+            throw new IllegalArgumentException("Cannot use " + getClass().getSimpleName() +
```

Comment:
It's a limitation of the PartitionCoalescer which we are duplicating from Spark. RangePartitionCoalescer is more restrictive, so we have a check to ensure it doesn't change the number of partitions.